wit/exit.wit

```diff
@@ -1,4 +1,12 @@
 interface exit {
   /// Exit the current instance and any linked instances.
   exit: func(status: result);
+
+  /// Exit the current instance and any linked instances, reporting the
+  /// specified status code to the host.
```

Comment:
Maybe add the sentence "This behavior is analogous to a trap, but without the connotation that something has bad happened."?